src/Table/index.tsx

```diff
@@ -0,0 +1,136 @@
+  s: CSSValue;
+  m: CSSValue;
+  l: CSSValue;
+};
+
+interface TableProps {
+  children: React.ReactNode;
+  /**
+   * Array describing when each column should be visible.
+   * If omitted, all columns will be visible regardless of viewport size.
+   *
+   * @usage
+   * First and last col always shown, middle col shown at "m" and up:
+   * `[ "*", "m", "*" ]`
+   */
+  colVisibility: ColMinBreakpoint[];
+  /**
+   * Specify a function that returns a `grid-template-columns` CSS value for each breakpoint.
+   * These are "mobile-first", so "m" means "the browser is at m or larger".
+   *
+   * Each function is provided with a `cols` argument.
+   */
+  colLayout?: ColLayoutConfig;
+  rowDensity?: "default" | "compact";
+}
+
+export const DEFAULT_COLS = 5;
+
+// By default all columns are shown
+export const defaultColVisibility = [
+  ...Array(DEFAULT_COLS).fill("*"),
+] as ColMinBreakpoint[];
+
+/**
+ * ⚠️ IN ACTIVE DEVELOPMENT ⚠️
+ * Availalbe for import, but not ready for production.
+ * Please leave storybook stories commented out until `@narmi/platform-ui` says so.
+ */
+const Table = ({
+  children,
+  colVisibility = defaultColVisibility,
+  colLayout = { s: "auto", m: "auto", l: "auto" },
+  rowDensity = "default",
+}: TableProps) => {
+  const { m, l } = useBreakpoints();
+  const currentBreakpoint = findCurrentFromBreakpoints({ m, l });
+
+  const visibleCols: number = colVisibility.filter(
+    (minRequired: ColMinBreakpoint) =>
+      isBreakpointSatisfied(
+        minRequired,
+        currentBreakpoint as ViewportBreakpoint,
+      ),
+  ).length;
+
+  const defaultLayout = {
+    s: `repeat(${visibleCols}, 1fr)`,
+    m: `repeat(${visibleCols}, 1fr)`,
+    l: `repeat(${visibleCols}, 1fr)`,
+  };
+
+  const validLayoutsFromProps = Object.fromEntries(
+    Object.entries(colLayout).filter(
+      ([, cssVal]) =>
+        // In browser environments, the CSS global has a `supports` method intended
+        // for feature detection. It also returns false if the property name
+        // or value passed in is invalid, acting as a validator.
+        CSS?.supports("grid-template-columns", cssVal) ?? true,
+    ),
+  );
+
+  const finalLayout = { ...defaultLayout, ...validLayoutsFromProps };
+  const gridTemplateColumns = columnTemplateFromBreakpoints(
+    currentBreakpoint as ViewportBreakpoint,
+    finalLayout,
+    visibleCols,
+  );
+
+  return (
+    <ColVisibilityContext.Provider
+      value={{
+        colVisibility,
+        currentBreakpoint: currentBreakpoint as ViewportBreakpoint,
+      }}
+    >
+      <div
+        role="table"
+        className={cc(["nds-table", `nds-table--${rowDensity}`])}
+        /**
+         * We apply a CSS value for grid-tempalte-columns on the root
+         * of this table component, so the grid column tracks may be
+         * shared among all rows via `subgrid`.
+         */
+        style={{ gridTemplateColumns }}
```

Comment:
This is where the magic happens. Through some layers of validation above, we determine a CSS value for `grid-template-columns` and set it on the table root element. 

Defining the table columns on root allows cells from different table sections (header, body) to participate in and share the same grid tracks via `subgrid`.